### PR TITLE
Routing documentation

### PR DESCRIPTION
**Pull request by itkg-rboessel**

### CODE REVIEW

File: en/developer_guide/routing.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
+Routing
+=======
+
+Base URL
+--------
+
+The page base url is compose with domain name and language prefix.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: In Open Orchestra, the base url of a website is composed by the domain and eventually a language prefix

File: en/developer_guide/routing.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+To manage all possibilities, Open Orchestra associate each website in database with aliases that contain:
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: in the database

Comment (nicolasThal) [inline en/developer_guide/routing.rst]: with website aliases

Comment (nicolasThal) [inline en/developer_guide/routing.rst]: which contains

File: en/developer_guide/routing.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+* **domain**: domain name
+* **language**: language prefix
+* **scheme** : to choose with which protocol page can be access
+
+Here is an example of some base URL generated for a website associate to two domains ``dev.example.com`` and ``test.example.com`` with english and french languages and default languages english.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: can you split the line?

Comment (nicolasThal) [inline en/developer_guide/routing.rst]: with english as default language and french as secondary one

File: en/developer_guide/routing.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
++----------+--------+------------------+----------+-----------------------------+
+| Alias Id | Scheme | Domain           | Language | Base URL                    |
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: this array seems wrong to me, it should also specify which are the language prefix for each alias

File: en/developer_guide/routing.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+=======
+
+Base URL
+--------
+
+The page base url is compose with domain name and language prefix.
+To manage all possibilities, Open Orchestra associate each website in database with aliases that contain:
+
+* **domain**: domain name
+* **language**: language prefix
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: There are two notion, the language which is the language in which all page are accessed and the prefix which is used to build the base url

File: en/developer_guide/routing.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+| 2        | http   | test.example.com | fr       | http://test.example.com     |
++----------+--------+------------------+----------+-----------------------------+
+| 3        | https  | test.example.com | en       | https://test.example.com/en |
++----------+--------+------------------+----------+-----------------------------+
+
+
+URL Pattern
+-----------
+
+The url pattern attribute is generated automatically with the name of the page or define by user for redirection.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: the end of the sentence seems wrong to me, can you rephrase it

File: en/developer_guide/routing.rst [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+It represents the pattern that will be the URL of the page or redirection on the website.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: I don't think that you should talk about redirection for now on

File: en/developer_guide/routing.rst [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+Implementation
+--------------
+
+``open-orchestra-front-bundle`` contain all routing files in the ``routing`` folder.
+
+
+Loader
+~~~~~~
+
+On Open Orchestra, ``Loader`` provide route from database collections:
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: In Open Orchestra

Comment (nicolasThal) [inline en/developer_guide/routing.rst]: What did you mean with `Loader`

File: en/developer_guide/routing.rst [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+* ``DatabaseRouteLoader`` generate routes for different nodes.
+Routes in loader are link with ``NodeController`` and take languages, node, site and alias in parameters.
+Route name is generate by two id : ``AliasID``_``NodeId``.
+
+* ``RedirectionLoader``generate routes for different redirection.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: If you talk about redirection, I think that you could introduce them in a paragraph just after the urlPattern one

File: en/developer_guide/routing.rst [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+``open-orchestra-front-bundle`` contain all routing files in the ``routing`` folder.
+
+
+Loader
+~~~~~~
+
+On Open Orchestra, ``Loader`` provide route from database collections:
+
+* ``DatabaseRouteLoader`` generate routes for different nodes.
+Routes in loader are link with ``NodeController`` and take languages, node, site and alias in parameters.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: Routes generated are linked to `NodeController` and contains the alias language, the node id, the site id and the alias id

File: en/developer_guide/routing.rst [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+Route name is generate by two id : ``AliasID``_``NodeId``.
+
+* ``RedirectionLoader``generate routes for different redirection.
+Routes in loader are link with ``FrameworkBundle:Redirect:urlRedirect`` for url redirection or ``FrameworkBundle:Redirect:redirect`` if their a node redirection in parameter.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: Can you split the line?

Comment (nicolasThal) [inline en/developer_guide/routing.rst]: Routes are linked to the redirect controller from the frameworkBundle

File: en/developer_guide/routing.rst [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+Route name is generate by two id : ``AliasID``_``RedirectionID``.
+
+See `Symfony Loader documentation`_ for more information.
+
+URL Generator
+~~~~~~~~~~~~~
+
+``OpenOrchestraUrlGenerator`` generate URL from route.
+It is use to take ``RequestStack`` in argument and get ``aliasId`` from master request and use this one to generate prefix for route name to adapt with loader.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: Can you rephrase this sentence?

File: en/developer_guide/routing.rst [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+
+See _`Request Stack documentation`_ for more information.
+
+Generator dumper
+~~~~~~~~~~~~~~~~
+
+``OpenOrchestraGeneratorDumper`` extend ``PhpGeneratorDumper`` which generate cache for route and avoid to request database for each connection.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: I think that you need a verb somewhere before which

File: en/developer_guide/routing.rst [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+=======
+
+Base URL
+--------
+
+In Open Orchestra, the base url of a website is composed by the domain and eventually a language prefix.
+To manage all possibilities, Open Orchestra associate each website in the database with website aliases which contain:
+
+* **domain**: domain name
+* **language prefix**: use by Open Orchestra to define the language
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: used

File: en/developer_guide/routing.rst [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+* **scheme** : to choose with which protocol page can be access
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: accessed

File: en/developer_guide/routing.rst [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+Here is an example of some base URL generated for a website associate to two domains: ``dev.example.com`` and ``test.example.com``.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: for a website associated with

File: en/developer_guide/routing.rst [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+| 2        | http   | test.example.com | fr       | fr              | http://test.example.com/fr  |
++----------+--------+------------------+----------+-----------------+-----------------------------+
+| 3        | https  | test.example.com | en       | none            | https://test.example.com    |
++----------+--------+------------------+----------+-----------------+-----------------------------+
+
+
+URL Pattern
+-----------
+
+The url pattern attribute is generated automatically with the name of the page.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: You were talking about website configuration and now about node configuration, but you didn't mentionned it. I think you should explicitly tell that you're changing subject.

Comment (ngilain) [inline en/developer_guide/routing.rst]: Moreover, the url pattern can be chosen by the contributor, it is only generated if the contributor doesn't give one :)

File: en/developer_guide/routing.rst [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+It represents the pattern that will be the URL of the page.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: "that will generate" more than "be" I think, as the url appends the prefix and the domain, and replaces dynamic parts of the pattern with real values

File: en/developer_guide/routing.rst [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+See ``Url pattern`` chapter from `Node Configuration`_ for more details.
+
+Redirection
+-----------
+
+Open Orchestra allows user to determine redirection in the back office.
+The redirection form contains some arguments:
+
+* **site name**: site to affect redirection
+* **site language**: language to affect redirection
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: What do you mean ? Is this related to the from url or the to url ?

File: en/developer_guide/routing.rst [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+* **pattern to redirect**: pattern that will be the URL of the redirection
```
Comment (ngilain) [inline en/developer_guide/routing.rst:51]: same question but I guess it's the from

File: en/developer_guide/routing.rst [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+* **node where the redirection is headed**: redirect to a node
+* **Url to redirect to**: redirect to url if node is not specify
+* **permanent redirection**: Use to adapt response code
+
+In this case, the URL Pattern is choose by user and not automatically generate.
+
+Implementation
+--------------
+
+``open-orchestra-front-bundle`` contain all routing files in the ``routing`` folder.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: contains

File: en/developer_guide/routing.rst [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+
+
+Loader
+~~~~~~
+
+In Open Orchestra, ``Route Loader`` provide route from database collections:
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: provides

File: en/developer_guide/routing.rst [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+
+* ``DatabaseRouteLoader`` generate routes for different nodes.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: generates

File: en/developer_guide/routing.rst [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+Routes in loader are link with ``NodeController`` and take languages, node, site and alias in parameters.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: are linked

File: en/developer_guide/routing.rst [hunk 25]
```diff
@@ -0,0 +1,95 @@
...
+Route name is generate by two id : ``AliasID``_``NodeId``.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: generated

File: en/developer_guide/routing.rst (hunk 9 above)
Comment (ngilain) [inline en/developer_guide/routing.rst]: redirections

File: en/developer_guide/routing.rst (hunk 11 above)
Comment (itkg-ngilain) [inline en/developer_guide/routing.rst]: are linked

Comment (ngilain) [inline en/developer_guide/routing.rst]: if there is a node redirection ?

File: en/developer_guide/routing.rst [hunk 26]
```diff
@@ -0,0 +1,95 @@
...
+
+* ``RedirectionLoader``generate routes for different redirection.
+Routes in loader are link with ``FrameworkBundle:Redirect:urlRedirect`` for url redirection or ``FrameworkBundle:Redirect:redirect`` if their a node redirection in parameter.
+Route name is generate by two id : ``AliasID``_``RedirectionID``.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: generated

File: en/developer_guide/routing.rst [hunk 27]
```diff
@@ -0,0 +1,95 @@
...
+
+See `Symfony Loader documentation`_ for more information.
+
+URL Generator
+~~~~~~~~~~~~~
+
+``OpenOrchestraUrlGenerator`` generate URL from route.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: generates

File: en/developer_guide/routing.rst [hunk 28]
```diff
@@ -0,0 +1,95 @@
...
+It is use to take ``RequestStack`` in argument and get ``aliasId`` from master request and use this one to generate prefix for route name to adapt with loader.
+
+See _`Request Stack documentation`_ for more information.
+
+Generator dumper
+~~~~~~~~~~~~~~~~
+
+``OpenOrchestraGeneratorDumper`` extend ``PhpGeneratorDumper`` which generate cache for route and avoid to request database for each connection.
+It override ``dump`` and ``generateGenerateMethod`` to add ``RequestStack```and ``AliasID`` in the cache to be coherent with other previous modifications.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: overrides

File: en/developer_guide/routing.rst [hunk 29]
```diff
@@ -0,0 +1,95 @@
...
+
+Router
+~~~~~~
+
+``OpenOrchestraRouter`` extend ``Symfony\Bundle\FrameworkBundle\Routing\Router`` to add ``RequestStack`` in argument for OpenOrchestraUrlGenerator Constructor.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: extends

**Comment by itkg-rboessel** [comment]
@nicolasThal @itkg-ngilain @itkg-nanne

File: en/developer_guide/routing.rst [hunk 30]
```diff
@@ -0,0 +1,92 @@
...
++----------+--------+------------------+----------+-----------------+-----------------------------+
+| 3        | https  | test.example.com | en       | none            | https://test.example.com    |
++----------+--------+------------------+----------+-----------------+-----------------------------+
+
+
+URL Pattern
+-----------
+
+The url pattern is added to the ``base URL`` to generate the url.
+For node, the url pattern can be choose by contributor or it is generated automatically from the page title.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: can be chosen

Comment (nicolasThal) [inline en/developer_guide/routing.rst]: by the contributor

Comment (nicolasThal) [inline en/developer_guide/routing.rst]: or is generated

File: en/developer_guide/routing.rst [hunk 31]
```diff
@@ -0,0 +1,92 @@
...
+* **site language**: contextual language which trigger redirection
+* **pattern to redirect**: pattern that will be the URL of the redirection
+* **node where the redirection is headed**: redirect to a node
+* **Url to redirect to**: redirect to url if node is not specify
+* **permanent redirection**: Use to adapt response code
+
+Implementation
+--------------
+
+``open-orchestra-front-bundle`` contains all routing files in the ``routing`` folder.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: ... all the routing related class in ....

File: en/developer_guide/routing.rst [hunk 32]
```diff
@@ -0,0 +1,92 @@
...
+
+
+Route Loader
+~~~~~~~~~~~~
+
+In Open Orchestra, ``Route Loader`` provide route from database collections:
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: provite a route collection extracted from the nodes

Comment (nicolasThal) [inline en/developer_guide/routing.rst]: put the real namespace of the class and note `Route Loader`

Comment (nicolasThal) [inline en/developer_guide/routing.rst]: After reading the rest of the paragraph, il think that you should rephrase this sentence

File: en/developer_guide/routing.rst [hunk 33]
```diff
@@ -0,0 +1,92 @@
...
+
+* ``DatabaseRouteLoader`` generate routes for different nodes.
+Routes in loader are linked with ``NodeController`` and take languages, node, site and alias in parameters.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: Routes generated are linked to the `NodeController`

Comment (nicolasThal) [inline en/developer_guide/routing.rst]: and have ... has parameters

File: en/developer_guide/routing.rst [hunk 34]
```diff
@@ -0,0 +1,92 @@
...
+Route name is generated by two id : ``AliasID``_``NodeId``.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst:87]: The route name generated follows this rule : `AliasId`_`NodeId`

File: en/developer_guide/routing.rst [hunk 35]
```diff
@@ -0,0 +1,92 @@
...
+~~~~~~~~~~~~~
+
+``OpenOrchestraUrlGenerator`` generates URL from route.
+The URL Generator is override to take `RequestStack`_ in argument and get the ``aliasId`` from it.
+It will be used to adapt this route name with their generate by Open Orchestra ``Route Loader``.
+
+Generator dumper
+~~~~~~~~~~~~~~~~
+
+``OpenOrchestraGeneratorDumper`` extend ``PhpGeneratorDumper``, it generate cache for route and avoid to request database for each connection.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: it will dump the routing in the Symfony cache file to avoid any request to the database

File: en/developer_guide/routing.rst [hunk 36]
```diff
@@ -0,0 +1,92 @@
...
+It override ``dump`` and ``generateGenerateMethod`` to add ``RequestStack```and ``AliasID`` in the cache to be coherent with other previous modifications.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: I am not sure I understood the end of the sentence, what did you mean?

File: en/developer_guide/routing.rst [hunk 37]
```diff
@@ -0,0 +1,92 @@
...
+
+Router
+~~~~~~
+
+``OpenOrchestraRouter`` extends ``Symfony\Bundle\FrameworkBundle\Routing\Router`` to add ``RequestStack`` in argument for OpenOrchestraUrlGenerator Constructor.
+
+.. _Node Configuration: /en/user_guide/node_configuration.rst
+.. _Symfony Loader documentation: http://symfony.com/doc/current/cookbook/routing/custom_route_loader.html
+.. _RequestStack: http://symfony.com/blog/new-in-symfony-2-4-the-request-stack
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: eof

File: en/developer_guide/routing.rst [hunk 38]
```diff
@@ -0,0 +1,92 @@
...
+Generator dumper
+~~~~~~~~~~~~~~~~
+
+``OpenOrchestraGeneratorDumper`` extend ``PhpGeneratorDumper``, it generate cache for route and avoid to request database for each connection.
+It override ``dump`` and ``generateGenerateMethod`` to add ``RequestStack```and ``AliasID`` in the cache to be coherent with other previous modifications.
+
+Router
+~~~~~~
+
+``OpenOrchestraRouter`` extends ``Symfony\Bundle\FrameworkBundle\Routing\Router`` to add ``RequestStack`` in argument for OpenOrchestraUrlGenerator Constructor.
```
Comment (nicolasThal) [inline en/developer_guide/routing.rst]: to add the `RequestStack` in the OpenOrchestraUrlGenerator constructor

File: en/developer_guide/routing.rst [hunk 39]
```diff
@@ -0,0 +1,110 @@
+Routing
+=======
+
+Base URL
+--------
+
+In Open Orchestra, the base url of a website is composed by the domain and eventually a language prefix.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: Eventually is not a transparent word. I guess you mean "possibly" or "potentially".

File: en/developer_guide/routing.rst [hunk 40]
```diff
@@ -0,0 +1,110 @@
...
+To manage all possibilities, Open Orchestra associate each website in the database with website aliases which contain:
```
Comment (juchi) [inline en/developer_guide/routing.rst]: all the possibilities
associates

File: en/developer_guide/routing.rst [hunk 41]
```diff
@@ -0,0 +1,110 @@
...
+
+* **domain**: domain name
+* **language prefix**: used by Open Orchestra to define the language
+* **scheme** : to choose with which protocol page can be accessed
+
+Here is an example of some base URL generated for a website associate to two domains: ``dev.example.com`` and ``test.example.com``.
+Each of them, with english as default language and french as secondary one.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: This sentence has no verb.

File: en/developer_guide/routing.rst (hunk 30 above)
Comment (juchi) [inline en/developer_guide/routing.rst]: "For a node" or "For nodes" depending on what you mean.

File: en/developer_guide/routing.rst [hunk 42]
```diff
@@ -0,0 +1,110 @@
...
+
+URL Pattern
+-----------
+
+The url pattern is added to the ``base URL`` to generate the url.
+For node, the url pattern can be choose by contributor or it is generated automatically from the page title.
+
+Url Pattern use Symfony rules :
+
+* Variables to exctract between {}, ex : `/exemple/{param}/test`
```
Comment (juchi) [inline en/developer_guide/routing.rst]: extract

File: en/developer_guide/routing.rst [hunk 43]
```diff
@@ -0,0 +1,110 @@
...
++----------+--------+------------------+----------+-----------------+-----------------------------+
+
+
+URL Pattern
+-----------
+
+The url pattern is added to the ``base URL`` to generate the url.
+For node, the url pattern can be choose by contributor or it is generated automatically from the page title.
+
+Url Pattern use Symfony rules :
```
Comment (juchi) [inline en/developer_guide/routing.rst]: "The url pattern uses" or "it uses" as there are some repetitions

File: en/developer_guide/routing.rst [hunk 44]
```diff
@@ -0,0 +1,110 @@
...
+
+* Variables to exctract between {}, ex : `/exemple/{param}/test`
+* Pattern have to be unique
+
+See ``Url pattern`` chapter from `Node Configuration`_ for more details.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: see the url pattern chapter

File: en/developer_guide/routing.rst [hunk 45]
```diff
@@ -0,0 +1,110 @@
...
+
+Redirection
+-----------
+
+Open Orchestra allows user to determine redirection in the back office.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: "allows a user" or "allows users"
"to determine redirection" is not clear to me. Shouldn't we explain briefly the functionnal behavior ?

File: en/developer_guide/routing.rst [hunk 46]
```diff
@@ -0,0 +1,110 @@
...
+The redirection form contains some arguments:
+
+* **site name**: contextual site which trigger redirection
```
Comment (juchi) [inline en/developer_guide/routing.rst]: triggers the redirection

File: en/developer_guide/routing.rst [hunk 47]
```diff
@@ -0,0 +1,110 @@
...
+* **site language**: contextual language which trigger redirection
```
Comment (juchi) [inline en/developer_guide/routing.rst]: idem

File: en/developer_guide/routing.rst [hunk 48]
```diff
@@ -0,0 +1,110 @@
...
+* **pattern to redirect**: pattern that will be the URL of the redirection
+* **node where the redirection is headed**: redirect to a node
+* **Url to redirect to**: redirect to url if node is not specify
```
Comment (juchi) [inline en/developer_guide/routing.rst:53]: "if a node is not specified" ?

File: en/developer_guide/routing.rst [hunk 49]
```diff
@@ -0,0 +1,110 @@
...
+* **permanent redirection**: Use to adapt response code
```
Comment (juchi) [inline en/developer_guide/routing.rst:54]: remove the capital U
"used"

File: en/developer_guide/routing.rst [hunk 50]
```diff
@@ -0,0 +1,110 @@
...
+
+Implementation
+--------------
+
+``open-orchestra-front-bundle`` contains all the routing related class in the ``routing`` folder.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: classes

File: en/developer_guide/routing.rst [hunk 51]
```diff
@@ -0,0 +1,110 @@
...
+
+
+Route Loader
+~~~~~~~~~~~~
+
+In Open Orchestra, route collections are extracted from the nodes by some classes:
+
+* ``DatabaseRouteLoader`` generate routes for different nodes.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: generates

File: en/developer_guide/routing.rst [hunk 52]
```diff
@@ -0,0 +1,110 @@
...
+Routes generated are linked to the ``NodeController`` and have languages, node, site and alias has parameters.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: Generated routes
has parameters -> as parameters ?

File: en/developer_guide/routing.rst [hunk 53]
```diff
@@ -0,0 +1,110 @@
...
+Route name is generated by two id : ``AliasID``_``NodeId``.
```
Comment (juchi) [inline en/developer_guide/routing.rst:87]: The route name is generated by two ids

File: en/developer_guide/routing.rst [hunk 54]
```diff
@@ -0,0 +1,110 @@
...
+
+* ``RedirectionLoader`` generate routes for different redirection.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: generates
redirections

File: en/developer_guide/routing.rst [hunk 55]
```diff
@@ -0,0 +1,110 @@
...
+Routes are linked to the redirect controller from the frameworkBundle.
+Route name is generated by two id : ``AliasID``_``RedirectionID``.
```
Comment (juchi) [inline en/developer_guide/routing.rst:91]: idem

File: en/developer_guide/routing.rst [hunk 56]
```diff
@@ -0,0 +1,110 @@
...
+
+Declare differents loaders as service in ``routing.yml`` is necessary to generate differents routes.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: Declaring
as services
"mandatory" instead of "necessary"
different

File: en/developer_guide/routing.rst [hunk 57]
```diff
@@ -0,0 +1,110 @@
...
+    open_orchestra_redirection:
+        resource: '.'
+        type: orchestra_redirection
+
+See `Symfony Loader documentation`_ for more information.
+
+URL Generator
+~~~~~~~~~~~~~
+
+``OpenOrchestraUrlGenerator`` generates URL from route.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: "a URL form a route" or "URLs form routes"

File: en/developer_guide/routing.rst [hunk 58]
```diff
@@ -0,0 +1,110 @@
...
+The URL Generator is override to take `RequestStack`_ in argument and get the ``aliasId`` from it.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: overriden
the aliasId of what ?

File: en/developer_guide/routing.rst [hunk 59]
```diff
@@ -0,0 +1,110 @@
...
+It will be used to adapt this route name with their generate by Open Orchestra ``Route Loader``.
```
Comment (juchi) [inline en/developer_guide/routing.rst]: "with their generate" i don't understand

File: en/developer_guide/routing.rst [hunk 60]
```diff
@@ -0,0 +1,110 @@
...
+
+Generator dumper
+~~~~~~~~~~~~~~~~
+
+``OpenOrchestraGeneratorDumper`` extend ``PhpGeneratorDumper``, it will dump the routing in the Symfony cache
```
Comment (juchi) [inline en/developer_guide/routing.rst]: extends

**Comment by itkg-rboessel** [comment]
@nicolasThal @itkg-nanne @itkg-ngilain

**Comment by nicolasThal** [comment]
:+1:

File: en/developer_guide/routing.rst [hunk 61]
```diff
@@ -0,0 +1,128 @@
...
+* Variables to extract between {}, ex : `/exemple/{param}/test`
+* Pattern have to be unique
+
+See the ``Url pattern`` chapter from `Node Configuration`_ for more details.
+
+Redirection
+-----------
+
+Open Orchestra allows users to determine redirection in the back office. When user
+The redirection form contains some arguments:
```
Comment (ngilain) [inline en/developer_guide/routing.rst:47]: I think something lacks here : "When user"

File: en/developer_guide/routing.rst [hunk 62]
```diff
@@ -0,0 +1,128 @@
...
+        type: orchestra_redirection
+
+See `Symfony Loader documentation`_ for more information.
+
+URL Generator
+~~~~~~~~~~~~~
+
+``OpenOrchestraUrlGenerator`` generates URLs from routes.
+The URL Generator is overriden to take `RequestStack`_ in argument and get the ``aliasId`` from it.
+It will be used to adapt this route name with route names generate by Open Orchestra ``Route Loader``.
```
Comment (ngilain) [inline en/developer_guide/routing.rst]: generated by

**Comment by ngilain** [comment]
:+1: when the last typo is corrected :)
